### PR TITLE
fix: The number of component replicas cannot be set to 0

### PR DESCRIPTION
**Pull request by daviderli614**
The default number of replicas frontend is 1, meta is 1, and datanode is 3. When the (frontend|meta|datanode) replicas set to 0, it will be equal to the default number of replicas. The replicas type is int32 and the k8s native [*int32 type](https://github.com/kubernetes/kubernetes/blob/0c645922edcc06adff43c70c02fb56751364bbb5/staging/src/k8s.io/api/apps/v1/types.go#L376). Check `replicas== nil`, it means the replicas number is not set.

### CODE REVIEW

File: apis/v1alpha1/defaulting.go [hunk 1]
```diff
@@ -98,30 +104,33 @@ func (in *GreptimeDBCluster) SetDefaults() error {
...
 		defaultGreptimeDBClusterSpec.Frontend = &FrontendSpec{
 			ComponentSpec: ComponentSpec{
-				Replicas: 1,
 				Template: &PodTemplateSpec{},
 			},
 			Service: ServiceSpec{
 				Type: corev1.ServiceTypeClusterIP,
 			},
 		}
+		if in.Spec.Frontend.Replicas == nil || *in.Spec.Frontend.Replicas < 0 {
```
Comment (zyy17) [inline apis/v1alpha1/defaulting.go:113]: If the replicas < 0, it should throw an error.

File: apis/v1alpha1/greptimedbcluster_types.go [hunk 2]
```diff
@@ -274,9 +274,8 @@ type PodTemplateSpec struct {
 // ComponentSpec is the common specification for all components(frontend/meta/datanode).
 type ComponentSpec struct {
 	// The number of replicas of the components.
-	// +kubebuilder:validation:Minimum=0
```
Comment (zyy17) [inline apis/v1alpha1/greptimedbcluster_types.go:277]: the validation rule should >= 0.

Comment (daviderli614) [reply apis/v1alpha1/greptimedbcluster_types.go:277]: done

**Review by zyy17 — APPROVED** [review]
lgtm

**Pull request merged**